test/fn/edit-contacts.spec.js

```diff
@@ -0,0 +1,174 @@
+const { expect, assert } = require('chai');
+const rewire = require('rewire');
+const PouchDB = require('pouchdb-core');
+PouchDB.plugin(require('pouchdb-adapter-memory'));
+const fs = require('../../src/lib/sync-fs');
+const editContactsModule = rewire('../../src/fn/edit-contacts');
+const environment = require('../../src/lib/environment');
+const path = require('path');
+const sinon = require('sinon');
+
+let pouch;
+
+// specifying directory paths to use
+const editContactsPath = `data/edit-contacts`;
+const saveDocsDir = `${editContactsPath}/json_docs`;
+const expectedDocsDirNested = `${editContactsPath}/expected-json_docs/nested-columns`;
+const expectedDocsDirOneCol = `${editContactsPath}/expected-json_docs/one-column`;
+const expectedDocsDirAllCols = `${editContactsPath}/expected-json_docs/all-columns`;
+const filesToUpload = fs.recurseFiles(`${editContactsPath}/server-contact_docs`).filter(name => name.endsWith('.json'));
+const countFilesInDir = path => fs.fs.readdirSync(path).length;
+
+const docs = filesToUpload
+  .map(file => {
+    const doc = fs.readJson(file);
+    return doc;
+  });
+
+const uploadDocuments = (docs) => {
+  return pouch.bulkDocs(docs);
+};
+
+function compareDocuments(expectedDocsDir){
+  fs.recurseFiles(expectedDocsDir)
+      .map(file => fs.path.basename(file))
+      .forEach(file => {
+        const expected  = fs.readJson(`${expectedDocsDir}/${file}`);
+        const generated = fs.readJson(`${saveDocsDir}/${file}`);
+        delete generated._rev;
+        expect(expected).to.deep.eq(generated);
+      });
+}
+
+describe('edit-contacts', function() {
+
+  beforeEach(async () => {
+    pouch = new PouchDB('edit-contacts', { adapter: 'memory' });
+    await uploadDocuments(docs);
+    sinon.stub(environment, 'pathToProject').get(() => editContactsPath);
+    const pouchDb = sinon.stub();
+    pouchDb.returns(pouch);
+    editContactsModule.__set__('pouch', pouchDb);
+  });
+
+  afterEach(async () => {
+    pouch.destroy();
+    fs.deleteFilesInFolder(saveDocsDir);
+  });
+
+  it(`should do a top-down test well and add all available columns to the docs since they are not specified`, async function(){
+
+    await editContactsModule.execute();
+    assert.equal(countFilesInDir(saveDocsDir),
+                countFilesInDir(expectedDocsDirAllCols),
+                `Different number of files in ${saveDocsDir} and ${expectedDocsDirAllCols}.`);
+    compareDocuments(expectedDocsDirAllCols);
+  }); 
+
+  it(`should only add specified column names to the json docs`, async function(){
+
+    const parseResult = {
+      colNames: ['is_in_emnch'],
+      csvFiles: ['contact.csv'],
+      docDirectoryPath: path.resolve(environment.pathToProject, 'json_docs'),
+    };
+    const extraArgs = sinon.stub();
+    extraArgs.returns(parseResult);
+    editContactsModule.__set__('parseExtraArgs', extraArgs);
```

Comment:
Please refrain from __set__-ing any "moving" parts of your module, especially if you're not testing them somewhere else (which you're not, in this case).
Replace every time you `editContactsModule.__set__('parseExtraArgs'` with something like:
```
sinon.stub(environment, 'extraArgs').get(() => ['--columns=is_in_emnch', '--files=contact.csv']);
```

Applies to all other tests as well.